### PR TITLE
POST Validation tests for User.

### PR DESCRIPTION
**Pull request by JAidhy**
V1.4e/createUserMessage.json

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 1]
```diff
@@ -399,7 +403,347 @@ public void test_negative_response_mandatory_elements_payload(final String userP
...
+        log.debug("The value of the Delegate Payload : " + delegateDTO.inputPayload());
+        commonDelegate.test_expected_response_for_supplied_header(
+                delegateDTO,
+                getSnlSuccessVerifier(),
+                new SNLVerificationDTO(HttpStatus.BAD_REQUEST, "1004", "[$.userRequest.details.personPositionFrom: 2002/10/02 is an invalid date]", null));
+    }
+
+    @ParameterizedTest(name = "personAuthorisedSessionTypes Positive Test Scenario")
+    @CsvSource(value = {"Valid personAuthorisedSessionTypes, DS",
+            "Valid personAuthorisedSessionTypes, ADHOC",
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:539]: May be move this to a CSV File...

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 2]
```diff
@@ -28,6 +29,9 @@
 import org.springframework.http.HttpStatus;
 import org.springframework.test.context.ActiveProfiles;
 
+import java.io.IOException;
+import java.util.*;
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:33]: OPtimise your imports in Intellij using ctrl options O.

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 3]
```diff
@@ -399,7 +403,347 @@ public void test_negative_response_mandatory_elements_payload(final String userP
...
+    @CsvSource(value = {"Valid phone number 19Chars,personContactValue", "Valid phone number 20Chars,personContactValue",
+            "Valid phone number 0Chars,personContactValue"}, nullValues = "NIL")
+    public void test_positive_response_for_person_contact_phone_with_mandatory_elements_payload(final String personPhoneKey,
+                                                                                                String personContactValue) throws Exception {
+
+        this.setInputPayloadFileName("resources-by-user-optional-person-contact-phone.json");
+        String contactPhoneValue = UUID.randomUUID().toString() + UUID.randomUUID().toString();
+
+
+        if (personPhoneKey.equalsIgnoreCase("Valid phone number 19Chars")) {
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:419]: Ideally use a Switch Case Here...

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 4]
```diff
@@ -399,7 +403,347 @@ public void test_negative_response_mandatory_elements_payload(final String userP
...
+    @ParameterizedTest(name = "personContactEmail positive scenario")
+    @CsvSource(value = {"Valid email 100 Chars,personEmailValue", "Valid email 99Chars,personEmailValue",
+            "Valid email 0Chars,personEmailValue"}, nullValues = "NIL")
+    public void test_positive_response_for_person_contact_email_with_mandatory_elements_payload(final String personContactEmailKey,
+                                                                                                String personEmailValue) throws Exception {
+
+        this.setInputPayloadFileName("resources-by-user-optional-person-email.json");
+        String contactEmail = UUID.randomUUID().toString() + UUID.randomUUID().toString() + UUID.randomUUID().toString();
+
+        if (personContactEmailKey.equalsIgnoreCase("Valid email 100 Chars")) {
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:463]: Same Comment above...

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 5]
```diff
@@ -399,7 +403,347 @@ public void test_negative_response_mandatory_elements_payload(final String userP
...
+        this.setInputPayloadFileName("resources-by-user-optional-person-contact-phone.json");
+        String contactPhoneValue = UUID.randomUUID().toString() + UUID.randomUUID().toString();
+        personContactValue = contactPhoneValue.substring(0, 21);
+        generatePayloadWithRandomHMCTSIDAndField(personContactValue, "/user/post/");
+        DelegateDTO delegateDTO = buildDelegateDTO(getRelativeURL(),
+                createStandardPayloadHeader(), getHttpMethod(), getHttpSuccessStatus());
+        log.debug("The value of the Delegate Payload : " + delegateDTO.inputPayload());
+        commonDelegate.test_expected_response_for_supplied_header(
+                delegateDTO,
+                getSnlSuccessVerifier(),
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:450]: Bug Here ....The Verifier for a Negative Test should be a Negative Verifier....

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 6]
```diff
@@ -399,7 +403,347 @@ public void test_negative_response_mandatory_elements_payload(final String userP
...
+        this.setInputPayloadFileName("resources-by-user-optional-person-email.json");
+        String contactEmail = UUID.randomUUID().toString() + UUID.randomUUID().toString() + UUID.randomUUID().toString();
+        personEmailValue = contactEmail.substring(0, 101);
+        generatePayloadWithRandomHMCTSIDAndField(personEmailValue, "/user/post/");
+        DelegateDTO delegateDTO = buildDelegateDTO(getRelativeURL(),
+                createStandardPayloadHeader(), getHttpMethod(), getHttpSuccessStatus());
+        log.debug("The value of the Delegate Payload : " + delegateDTO.inputPayload());
+        commonDelegate.test_expected_response_for_supplied_header(
+                delegateDTO,
+                getSnlSuccessVerifier(),
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:495]: Negative Verifier....

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 7]
```diff
@@ -399,7 +403,347 @@ public void test_negative_response_mandatory_elements_payload(final String userP
...
+    public void test_negative_person_position_from_date_with_mandatory_elements_payload(final String validPersonPositionFromDate,
+                                                                                        final String validPersonPositionFromDateValue) throws IOException {
+        this.setInputPayloadFileName("resources-by-user-optional-person-position-from-date.json");
+        generatePayloadWithRandomHMCTSIDAndField(validPersonPositionFromDateValue, "/user/post/");
+        DelegateDTO delegateDTO = buildDelegateDTO(getRelativeURL(),
+                createStandardPayloadHeader(), getHttpMethod(), getHttpSuccessStatus());
+        log.debug("The value of the Delegate Payload : " + delegateDTO.inputPayload());
+        commonDelegate.test_expected_response_for_supplied_header(
+                delegateDTO,
+                getSnlSuccessVerifier(),
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:533]: Negative verifier....

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 8]
```diff
@@ -399,7 +403,347 @@ public void test_negative_response_mandatory_elements_payload(final String userP
...
+        generatePayloadWithRandomHMCTSIDAndPersonField(personAuthorisedSessionValue, "/user/post/",
+                "[\"personAuthorisedSessionTypes\"]");
+
+        DelegateDTO delegateDTO = buildDelegateDTO(getRelativeURL(),
+                createStandardPayloadHeader(), getHttpMethod(), getHttpSuccessStatus());
+        log.debug("The value of the Delegate Payload : " + delegateDTO.inputPayload());
+
+        commonDelegate.test_expected_response_for_supplied_header(
+                delegateDTO,
+                getSnlSuccessVerifier(),
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:602]: Negative Verifier...

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 9]
```diff
@@ -399,7 +403,347 @@ public void test_negative_response_mandatory_elements_payload(final String userP
...
+        }
+
+        for (int positionTitle = 153; positionTitle < 158; positionTitle++) {
+            generatePayloadWithRandomHMCTSIDAndField(String.valueOf(positionTitle), "/user/post/");
+            DelegateDTO delegateDTO = buildDelegateDTO(getRelativeURL(),
+                    createStandardPayloadHeader(), getHttpMethod(), getHttpSuccessStatus());
+            log.debug("The value of the Delegate Payload : " + delegateDTO.inputPayload());
+            commonDelegate.test_expected_response_for_supplied_header(
+                    delegateDTO,
+                    getSnlSuccessVerifier(),
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:666]: Negative Verifier....

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java [hunk 10]
```diff
@@ -399,7 +403,347 @@ public void test_negative_response_mandatory_elements_payload(final String userP
...
+        ObjectMapper objectMapper = new ObjectMapper();
+        String data = objectMapper.writeValueAsString(PersonJOHTypeArray);
+
+        generatePayloadWithRandomHMCTSIDAndPersonField(data, "/user/post/", "[\"personJOHTypes\"]");
+        DelegateDTO delegateDTO = buildDelegateDTO(getRelativeURL(),
+                createStandardPayloadHeader(), getHttpMethod(), getHttpSuccessStatus());
+        log.debug("The value of the Delegate Payload : " + delegateDTO.inputPayload());
+        commonDelegate.test_expected_response_for_supplied_header(
+                delegateDTO,
+                getSnlSuccessVerifier(),
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:744]: Negative Verifier...

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/ResourcesPayloadValidationTest.java [hunk 11]
```diff
@@ -100,4 +100,21 @@ final void generateLocationPayloadWithRandomHMCTSIDAndFieldTokenReplace(final St
                 getInputFileDirectory()) + templatePath + getInputPayloadFileName()), randomId, value);
         this.setInputBodyPayload(replaceCharacterSequence(token, value, formattedString));
     }
+
+    final void generatePayloadWithRandomHMCTSIDAndField(final Object formatValue, final String templatePath) throws IOException {
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/ResourcesPayloadValidationTest.java:104]: I am confident that we already have a method like this. Do we have to Duplicate?????

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/ResourcesPayloadValidationTest.java [hunk 12]
```diff
@@ -100,4 +100,21 @@ final void generateLocationPayloadWithRandomHMCTSIDAndFieldTokenReplace(final St
...
+        final String randomID = UUID.randomUUID().toString() + UUID.randomUUID().toString();
+        this.setInputBodyPayload(String.format(TestingUtils.readFileContents(String.format(INPUT_TEMPLATE_FILE_PATH,
+                getInputFileDirectory()) + templatePath + getInputPayloadFileName()), randomID, formatValue));
+
+    }
+
+    final void generatePayloadWithRandomHMCTSIDAndPersonField(final Object formatValue,
```
Comment (pats-john) [inline src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/ResourcesPayloadValidationTest.java:111]: What is the purpose of this util method...?Just need to discuss....

**Review by pats-john — CHANGES_REQUESTED** [review]

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java (hunk 1 above)
Comment (JAidhy) [reply src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:539]: Implemented using csv file.

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java (hunk 3 above)
Comment (JAidhy) [reply src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:419]: Implemented as suggested

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java (hunk 4 above)
Comment (JAidhy) [reply src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:463]: Implemented as suggested

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java (hunk 5 above)
Comment (JAidhy) [reply src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:450]: Implemented as suggested

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java (hunk 8 above)
Comment (JAidhy) [reply src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:602]: Implemented as suggested

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java (hunk 9 above)
Comment (JAidhy) [reply src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:666]: Implemented as suggested

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java (hunk 10 above)
Comment (JAidhy) [reply src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/POSTResourcesByUserPayloadValidationTest.java:744]: Implemented as suggested

File: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/ResourcesPayloadValidationTest.java (hunk 11 above)
Comment (JAidhy) [reply src/acceptanceTest/java/uk/gov/hmcts/futurehearings/snl/acceptance/resources/ResourcesPayloadValidationTest.java:104]: Not using this method anymore, I have commented it.

**Review by pats-john — APPROVED** [review]

**Pull request merged**